redfish-core/lib/pcie.hpp

```diff
@@ -500,6 +500,88 @@ inline void
     });
 }
 
+inline void
+    getPCIeDevicePrettyName(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                            const std::string& pcieDevicePath,
+                            const std::string& service)
+{
+    sdbusplus::asio::getAllProperties(
+        *crow::connections::systemBus, service, pcieDevicePath,
+        "xyz.openbmc_project.Inventory.Item",
+        [pcieDevicePath, asyncResp{asyncResp}](
+            const boost::system::error_code& ec,
+            const dbus::utility::DBusPropertiesMap& itemList) {
+        if (ec)
+        {
+            if (ec.value() != EBADR)
+            {
+                BMCWEB_LOG_ERROR("DBUS response error for Properties{}",
+                                 ec.value());
+                messages::internalError(asyncResp->res);
+            }
+            return;
+        }
+
+        const std::string* prettyName = nullptr;
+
+        const bool success = sdbusplus::unpackPropertiesNoThrow(
+            dbus_utils::UnpackErrorPrinter(), itemList, "PrettyName",
+            prettyName);
+
+        if (!success)
+        {
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        if (prettyName != nullptr)
+        {
+            asyncResp->res.jsonValue["Name"] = *prettyName;
+        }
+    });
+}
+
+inline void
+    getPCIeDeviceLocation(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                          const std::string& pcieDevicePath,
+                          const std::string& service)
+{
+    sdbusplus::asio::getAllProperties(
+        *crow::connections::systemBus, service, pcieDevicePath,
+        "xyz.openbmc_project.Inventory.Decorator.LocationCode",
+        [pcieDevicePath, asyncResp{asyncResp}](
```

Comment:
Is it sufficient to use [asyncResp] (instead of [asyncResp{asyncResp}])?

Reply:
yes, done